docs/cloud_setup_mqtt.md

```diff
@@ -0,0 +1,165 @@
+      # WRITABLE_PIPELINE_FUNCTIONS_MQTTSEND_PARAMETERS_QOS: ["your quality or service"]
+      # WRITABLE_PIPELINE_FUNCTIONS_MQTTSEND_PARAMETERS_KEY: [your Key]
+      # WRITABLE_PIPELINE_FUNCTIONS_MQTTSEND_PARAMETERS_CERT: [your Certificate]
+    depends_on:
+      - consul
+      # - logging  # uncomment if re-enabled remote logging
+      - data
+```
+
+For more details please refer to the [Official Guide](https://docs.edgexfoundry.org/1.3/getting-started/quick-start/#exporting-data).
+
+## 3. Getting Started With CloudMQTT
+
+CloudMQTT are managed Mosquitto servers in the cloud. Mosquitto implements the MQ Telemetry Transport protocol, MQTT, which provides lightweight methods of carrying out messaging using a publish/subscribe message queueing model.
+
+![Producer Consumer](images/CloudSync/ProducerConsumer.png)
+
+MQTT is the machine-to-machine protocol of the future. It is ideal for the “Internet of Things” world of connected devices. Its minimal design makes it perfect for built-in systems, mobile phones and other memory and bandwidth sensitive applications.
+
+Message queues provide an asynchronous communications protocol, the sender and receiver of the message do not need to interact with the message queue at the same time. Messages placed onto the queue are stored until the recipient retrieves them or until the messages times out. MQTT and Mosquitto are for good use by bandwidth sensitive applications.
+
+CloudMQTT let you focus on the application instead of spending time on scaling the broker or patching the platform.
+
+![Role of MQTT](images/CloudSync/RoleOfMQTT.png)
+
+## 4. CloudMQTT Setup on Heroku
+
+#### For Basic Understanding and setup of Heroku CLoud please visit https://devcenter.heroku.com/
+
+#### Setting Up CloudMQTT 
+CloudMQTT is an add-on for providing a MQTT broker to your application(s).
+MQTT is a lightweight pub/sub protocol, especially suited for low processor/bandwidth units like sensors and built-in system, but also suited for fast communication within applications.
+
+#### Provisioning the add-on
+```
+heroku addons:create cloudmqtt
+-----> Adding cloudmqtt to sharp-mountain-4005... done, v18 (free)
+```
+
+Once CloudMQTT has been added a `CLOUDMQTT_URL` setting will be available in the app configuration and will contain the canonical URL used to access the newly provisioned CloudMQTT service instance. This can be confirmed using the `heroku config:get` command.
+
+```
+heroku config:get CLOUDMQTT_URL
+mqtt://user:pass@broker.cloudmqtt.com
+```
+
+#### Dashboard
+The CloudMQTT dashboard allows you to monitor your current connection and message traffic statistics.
+
+The dashboard can be accessed via the CLI:
+```
+heroku addons:open cloudmqtt
+Opening cloudmqtt for sharp-mountain-4005...
+```
+or by visiting the [Heroku Dashboard](https://dashboard.heroku.com/) and selecting the application in question. Select CloudMQTT from the Add-ons menu.
+
+# 5. Optional - Testing Setup on A Public Broker - HiveMQ
```

Comment:
This is a very useful guide, but the contents (including pictures) seem to be derived from edgex. 
If right, how about guiding this to its original link? You can put short comments with the link on this document.

Reply:
@t25kim Some parts (Like point 5) have been derived from EdgeX. Other parts have been written by me. Some images have also been drawn by me, while others as in point 5 have been derived. Would you like that I add the link to references (Same as point 3), or remove the data and add the link only ? I personally feel It is better to leave the content for easy readability and continuity.

Reply:
I agree that it is very helpful in terms of readability and continuity. 
What about modifying `HiveMQTT Messages` and `Subscriptions` pictures in accordance with Home Edge?
I think they should be revised to fit our project. If the modification is difficult, leave the link along with brief descriptions.

Reply:
Done.